snap/naming/validate.go

```diff
@@ -332,6 +334,37 @@ func ValidateAssumes(assumes []string, currentSnapdVersion string, featureSet ma
 			}
 		}
 
+		if strings.HasPrefix(flag, "isa") {
```

Comment:
Should be like this so we don't match things that otherwise would start with `isa`
```suggestion
		if strings.HasPrefix(flag, "isa-") {
```